### ISSUE TITLE
Picked color is always FFFFFF when using graphicsmagick-imagemagick-compat

### ISSUE THREAD

**Issue by jghauser** [jgmdev/wl-color-picker#1]
Great tool! I've been looking for a color picker that works on wayland/wlroots/sway for a while now. When I tried using your tool, the resulting color seemed to always be FFFFFF (just to make sure, the image I'm sampling from isn't white). A little more testing revealed that this only happens with graphicsmagick and graphicsmagick-imagemagick-compat installed. It works with imagemagick. I installed the package from AUR and imagemagick wasn't installed as a dependency as graphicsmagick-imagemagick-compat already provides that.

Maybe this is more of an issue with graphicsmagick-imagemagick-compat, as this package clearly doesn't provide the necessary compatibility in this case.

It would be great if your tool did work in this case also. If that's not within the scope of this tool, maybe you'll find this report useful to inform other users in case they run into the same problem.

**Comment by jgmdev** [comment]
Thanks for sharing the information! I see that graphicsmagick is basically like a fork of imagemagick, so graphicsmagick-imagemagick-compat provides convert? if you provide a sample of what has to be done different to properly get the color value using graphicsmagick-imagemagick-compat I would add the neccesary check for convert binaries or version strings and execute a different command depending on what is available. Also if easier, feel free to send a pull request.

**Comment by jghauser** [comment]
I had some time to look into this now. I think the reason it didn't work before is that the output of `convert` isn't the same with graphicsmagick-imagemagick-compat in the case of text output. Here's the solution I came up with:

    if command -v /bin/gm &> /dev/null; then
      color=$(grim -g "$position" -t ppm - | /bin/gm convert - -format '%[pixel:p{0,0}]' txt:- | tail -n 1 | rev | cut -d ' ' -f 1 | rev)
    else
      color=$(grim -g "$position" -t ppm - | convert - -format '%[pixel:p{0,0}]' txt:- | tail -n 1 | cut -d ' ' -f 4)
    fi

Here's what it does:

- It checks whether the graphicsmagick executable exists, and if it does, it uses `gm convert` instead of `convert`. This means the script doesn't depend on imagemagick anymore, either graphicsmagick or imagemagick works.
- I used `/bin/gm` because on many systems (including mine),  `gm` is an alias for `git commit`
- I had to pipe things through `rev` because the relevant field isn't always the same one counted from the front, but is always the last one
